// ionic2-calendar/dayview.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@angular/common';
import * as import2 from './init-position-scroll.ngfactory';
import * as import3 from './init-position-scroll';
import * as import4 from './dayview';
import * as import5 from 'ionic-angular/components/slides/slides.ngfactory';
import * as import6 from 'ionic-angular/components/slides/slides';
import * as import7 from 'ionic-angular/config/config';
import * as import8 from 'ionic-angular/platform/platform';
import * as import9 from 'ionic-angular/navigation/view-controller';
import * as import10 from 'ionic-angular/components/slides/slide.ngfactory';
import * as import11 from 'ionic-angular/components/slides/slide';
import * as import12 from 'ionic-angular/components/scroll/scroll.ngfactory';
import * as import13 from 'ionic-angular/components/scroll/scroll';
import * as import14 from './calendar.service';
const styles_DayViewComponent:any[] = ['\n        .table-fixed {\n          table-layout: fixed;\n        }\n\n        .table {\n          width: 100%;\n          max-width: 100%;\n          background-color: transparent;\n        }\n\n        .table > thead > tr > th, .table > tbody > tr > th, .table > tfoot > tr > th, .table > thead > tr > td,\n        .table > tbody > tr > td, .table > tfoot > tr > td {\n          padding: 8px;\n          line-height: 20px;\n          vertical-align: top;\n        }\n\n        .table > thead > tr > th {\n          vertical-align: bottom;\n          border-bottom: 2px solid #ddd;\n        }\n\n        .table > thead:first-child > tr:first-child > th, .table > thead:first-child > tr:first-child > td {\n          border-top: 0\n        }\n\n        .table > tbody + tbody {\n          border-top: 2px solid #ddd;\n        }\n\n        .table-bordered {\n          border: 1px solid #ddd;\n        }\n\n        .table-bordered > thead > tr > th, .table-bordered > tbody > tr > th, .table-bordered > tfoot > tr > th,\n        .table-bordered > thead > tr > td, .table-bordered > tbody > tr > td, .table-bordered > tfoot > tr > td {\n          border: 1px solid #ddd;\n        }\n\n        .table-bordered > thead > tr > th, .table-bordered > thead > tr > td {\n          border-bottom-width: 2px;\n        }\n\n        .table-striped > tbody > tr:nth-child(odd) > td, .table-striped > tbody > tr:nth-child(odd) > th {\n          background-color: #f9f9f9\n        }\n\n        .calendar-hour-column {\n          width: 50px;\n          white-space: nowrap;\n        }\n\n        .calendar-event-wrap {\n          position: relative;\n          width: 100%;\n          height: 100%;\n        }\n\n        .calendar-event {\n          position: absolute;\n          padding: 2px;\n          cursor: pointer;\n          z-index: 10000;\n        }\n\n        .calendar-cell {\n          padding: 0 !important;\n          height: 37px;\n        }\n\n        .dayview-allday-label {\n          float: left;\n          height: 100%;\n          line-height: 50px;\n          text-align: center;\n          width: 50px;\n          border-left: 1px solid #ddd;\n        }\n\n        [dir="rtl"] .dayview-allday-label {\n            border-right: 1px solid #ddd;\n            float: right;\n        }\n\n        .dayview-allday-content-wrapper {\n          margin-left: 50px;\n          overflow: hidden;\n          height: 51px;\n        }\n\n        [dir="rtl"] .dayview-allday-content-wrapper {\n          margin-left: 0;\n          margin-right: 50px;\n        }\n\n        .dayview-allday-content-table {\n          min-height: 50px;\n        }\n\n        .dayview-allday-content-table td {\n          border-left: 1px solid #ddd;\n          border-right: 1px solid #ddd;\n        }\n\n        .dayview-allday-table {\n          height: 50px;\n          position: relative;\n          border-bottom: 1px solid #ddd;\n          font-size: 14px;\n        }\n\n        .dayview-normal-event-container {\n          margin-top: 50px;\n          overflow: hidden;\n          left: 0;\n          right: 0;\n          top: 0;\n          bottom: 0;\n          position: absolute;\n          font-size: 14px;\n        }\n\n        .dayview .slide-zoom {\n          height: 100%;\n        }\n\n        .dayview-allday-content-wrapper scroll-content {\n          width: 100%;\n        }\n\n        ::-webkit-scrollbar,\n        *::-webkit-scrollbar {\n          display: none;\n        }\n\n        .table > tbody > tr > td.calendar-hour-column {\n          padding-left: 0;\n          padding-right: 0;\n          vertical-align: middle;\n        }\n\n        @media (max-width: 750px) {\n          .dayview-allday-label, .calendar-hour-column {\n            width: 31px;\n            font-size: 12px;\n          }\n\n          .dayview-allday-label {\n            padding-top: 4px;\n          }\n\n          .table > tbody > tr > td.calendar-hour-column {\n            padding-left: 0;\n            padding-right: 0;\n            vertical-align: middle;\n            line-height: 12px;\n          }\n\n          .dayview-allday-label {\n            line-height: 20px;\n          }\n\n          .dayview-allday-content-wrapper {\n            margin-left: 31px;\n          }\n\n          [dir="rtl"] .dayview-allday-content-wrapper {\n            margin-left: 0;\n            margin-right: 31px;\n          }\n        }\n    '];
export const RenderType_DayViewComponent:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_DayViewComponent,
  data: {}
}
);
function View_DayViewComponent_3(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[(l()(),import0.ɵted((null as any),['\n                                        ']))],(null as any),(null as any));
}
function View_DayViewComponent_2(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),7,'div',[
      [
        'class',
        'calendar-event'
      ]
      ,
      [
        'tappable',
        ''
      ]

    ]
      ,(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.eventSelected(v.context.$implicit.event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import1.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    import0.ɵpod([
      'top',
      'width',
      'height'
    ]
    ),
    (l()(),import0.ɵted((null as any),['\n                                        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),2,(null as any),View_DayViewComponent_3)),
    import0.ɵdid(270336,(null as any),0,import1.NgTemplateOutlet,[import0.ViewContainerRef],{
      ngTemplateOutlet: [
        0,
        'ngTemplateOutlet'
      ]
      ,
      ngOutletContext: [
        1,
        'ngOutletContext'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['displayEvent']),
    (l()(),import0.ɵted((null as any),['\n                                    ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = ck(v,2,0,((25 * v.context.index) + 'px'),'100%','25px');
    ck(v,1,0,currVal_0);
    const currVal_1:any = co.dayviewAllDayEventTemplate;
    const currVal_2:any = ck(v,6,0,v.context.$implicit);
    ck(v,5,0,currVal_1,currVal_2);
  },(null as any));
}
function View_DayViewComponent_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),8,'td',[[
        'class',
        'calendar-cell'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import1.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['calendar-event-wrap']),
    import0.ɵdid(139264,(null as any),0,import1.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    import0.ɵpod(['height']),
    (l()(),import0.ɵted((null as any),['\n                                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_DayViewComponent_2)),
    import0.ɵdid(401408,(null as any),0,import1.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                                ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = 'calendar-cell';
    const currVal_1:any = ck(v,2,0,(co.views[0].allDayEvents.length > 0));
    ck(v,1,0,currVal_0,currVal_1);
    const currVal_2:any = ck(v,4,0,((25 * co.views[0].allDayEvents.length) + 'px'));
    ck(v,3,0,currVal_2);
    const currVal_3:any = co.views[0].allDayEvents;
    ck(v,7,0,currVal_3);
  },(null as any));
}
function View_DayViewComponent_4(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'td',[[
        'class',
        'calendar-cell'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                                ']))
  ]
  ,(null as any),(null as any));
}
function View_DayViewComponent_9(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[(l()(),import0.ɵted((null as any),['\n                                         ']))],(null as any),(null as any));
}
function View_DayViewComponent_8(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),7,'div',[
      [
        'class',
        'calendar-event'
      ]
      ,
      [
        'tappable',
        ''
      ]

    ]
      ,(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.eventSelected(v.context.$implicit.event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import1.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    import0.ɵpod([
      'top',
      'left',
      'width',
      'height'
    ]
    ),
    (l()(),import0.ɵted((null as any),['\n                                         '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),2,(null as any),View_DayViewComponent_9)),
    import0.ɵdid(270336,(null as any),0,import1.NgTemplateOutlet,[import0.ViewContainerRef],{
      ngTemplateOutlet: [
        0,
        'ngTemplateOutlet'
      ]
      ,
      ngOutletContext: [
        1,
        'ngOutletContext'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['displayEvent']),
    (l()(),import0.ɵted((null as any),['\n                                    ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = ck(v,2,0,(((37 * v.context.$implicit.startOffset) / co.hourParts) + 'px'),(((100 / v.context.$implicit.overlapNumber) * v.context.$implicit.position) + '%'),((100 / v.context.$implicit.overlapNumber) + '%'),((37 * ((v.context.$implicit.endIndex - v.context.$implicit.startIndex) - ((v.context.$implicit.endOffset + v.context.$implicit.startOffset) / co.hourParts))) + 'px'));
    ck(v,1,0,currVal_0);
    const currVal_1:any = co.dayviewNormalEventTemplate;
    const currVal_2:any = ck(v,6,0,v.context.$implicit);
    ck(v,5,0,currVal_1,currVal_2);
  },(null as any));
}
function View_DayViewComponent_7(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),6,'div',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import1.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngClass: [
        0,
        'ngClass'
      ]
    },(null as any)),
    import0.ɵpod(['calendar-event-wrap']),
    (l()(),import0.ɵted((null as any),['\n                                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_DayViewComponent_8)),
    import0.ɵdid(401408,(null as any),0,import1.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                                ']))
  ]
  ,(ck,v) => {
    const currVal_0:any = ck(v,2,0,(<any>v.parent).context.$implicit.events);
    ck(v,1,0,currVal_0);
    const currVal_1:any = (<any>v.parent).context.$implicit.events;
    ck(v,5,0,currVal_1);
  },(null as any));
}
function View_DayViewComponent_6(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),10,'tr',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'td',[[
        'class',
        'calendar-hour-column text-center'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '\n                                ',
      '\n                            '
    ]
    )),
    (l()(),import0.ɵted((null as any),['\n                            '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),4,'td',[
      [
        'class',
        'calendar-cell'
      ]
      ,
      [
        'tappable',
        ''
      ]

    ]
      ,(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.select(v.context.$implicit.time,v.context.$implicit.events)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_DayViewComponent_7)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                            '])),
    (l()(),import0.ɵted((null as any),['\n                        ']))
  ]
  ,(ck,v) => {
    const currVal_1:any = v.context.$implicit.events;
    ck(v,8,0,currVal_1);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.hourColumnLabels[v.context.index];
    ck(v,3,0,currVal_0);
  });
}
function View_DayViewComponent_5(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),11,'init-position-scroll',[[
        'class',
        'dayview-normal-event-container'
      ]
      ],(null as any),[[
        (null as any),
        'onScroll'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('onScroll' === en)) {
        const pd_0:any = ((<any>co.setScrollPosition($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },import2.View_initPositionScrollComponent_0,import2.RenderType_initPositionScrollComponent)),
    import0.ɵdid(2449408,(null as any),0,import3.initPositionScrollComponent,[import0.ElementRef],{
      initPosition: [
        0,
        'initPosition'
      ]
      ,
      emitEvent: [
        1,
        'emitEvent'
      ]

    }
    ,{onScroll: 'onScroll'}),
    (l()(),import0.ɵted(0,['\n                    '])),
      (l()(),import0.ɵeld(0,(null as any),0,7,'table',[[
        'class',
        'table table-bordered table-fixed dayview-normal-event-table'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),4,'tbody',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_DayViewComponent_6)),
    import0.ɵdid(401408,(null as any),0,import1.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵted(0,['\n                ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.initScrollPosition;
    const currVal_1:any = co.preserveScrollPosition;
    ck(v,1,0,currVal_0,currVal_1);
    const currVal_2:any = co.views[0].rows;
    ck(v,8,0,currVal_2);
  },(null as any));
}
function View_DayViewComponent_11(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),7,'tr',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'td',[[
        'class',
        'calendar-hour-column text-center'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '\n                                ',
      '\n                            '
    ]
    )),
    (l()(),import0.ɵted((null as any),['\n                            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'td',[[
        'class',
        'calendar-cell'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                            '])),
    (l()(),import0.ɵted((null as any),['\n                        ']))
  ]
  ,(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.hourColumnLabels[v.context.index];
    ck(v,3,0,currVal_0);
  });
}
function View_DayViewComponent_10(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),11,'init-position-scroll',[[
        'class',
        'dayview-normal-event-container'
      ]
    ],(null as any),(null as any),(null as any),import2.View_initPositionScrollComponent_0,import2.RenderType_initPositionScrollComponent)),
      import0.ɵdid(2449408,(null as any),0,import3.initPositionScrollComponent,[import0.ElementRef],{initPosition: [
        0,
        'initPosition'
      ]
    },(null as any)),
    (l()(),import0.ɵted(0,['\n                    '])),
      (l()(),import0.ɵeld(0,(null as any),0,7,'table',[[
        'class',
        'table table-bordered table-fixed dayview-normal-event-table'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),4,'tbody',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_DayViewComponent_11)),
    import0.ɵdid(401408,(null as any),0,import1.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵted(0,['\n                ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.initScrollPosition;
    ck(v,1,0,currVal_0);
    const currVal_1:any = co.views[0].rows;
    ck(v,8,0,currVal_1);
  },(null as any));
}
function View_DayViewComponent_14(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[(l()(),import0.ɵted((null as any),['\n                                        ']))],(null as any),(null as any));
}
function View_DayViewComponent_13(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),7,'div',[
      [
        'class',
        'calendar-event'
      ]
      ,
      [
        'tappable',
        ''
      ]

    ]
      ,(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.eventSelected(v.context.$implicit.event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import1.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    import0.ɵpod([
      'top',
      'width',
      'height'
    ]
    ),
    (l()(),import0.ɵted((null as any),['\n                                        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),2,(null as any),View_DayViewComponent_14)),
    import0.ɵdid(270336,(null as any),0,import1.NgTemplateOutlet,[import0.ViewContainerRef],{
      ngTemplateOutlet: [
        0,
        'ngTemplateOutlet'
      ]
      ,
      ngOutletContext: [
        1,
        'ngOutletContext'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['displayEvent']),
    (l()(),import0.ɵted((null as any),['\n                                    ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = ck(v,2,0,((25 * v.context.index) + 'px'),'100%','25px');
    ck(v,1,0,currVal_0);
    const currVal_1:any = co.dayviewAllDayEventTemplate;
    const currVal_2:any = ck(v,6,0,v.context.$implicit);
    ck(v,5,0,currVal_1,currVal_2);
  },(null as any));
}
function View_DayViewComponent_12(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),8,'td',[[
        'class',
        'calendar-cell'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import1.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['calendar-event-wrap']),
    import0.ɵdid(139264,(null as any),0,import1.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    import0.ɵpod(['height']),
    (l()(),import0.ɵted((null as any),['\n                                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_DayViewComponent_13)),
    import0.ɵdid(401408,(null as any),0,import1.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                                ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = 'calendar-cell';
    const currVal_1:any = ck(v,2,0,(co.views[1].allDayEvents.length > 0));
    ck(v,1,0,currVal_0,currVal_1);
    const currVal_2:any = ck(v,4,0,((25 * co.views[1].allDayEvents.length) + 'px'));
    ck(v,3,0,currVal_2);
    const currVal_3:any = co.views[1].allDayEvents;
    ck(v,7,0,currVal_3);
  },(null as any));
}
function View_DayViewComponent_15(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'td',[[
        'class',
        'calendar-cell'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                                ']))
  ]
  ,(null as any),(null as any));
}
function View_DayViewComponent_20(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[(l()(),import0.ɵted((null as any),['\n                                         ']))],(null as any),(null as any));
}
function View_DayViewComponent_19(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),7,'div',[
      [
        'class',
        'calendar-event'
      ]
      ,
      [
        'tappable',
        ''
      ]

    ]
      ,(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.eventSelected(v.context.$implicit.event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import1.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    import0.ɵpod([
      'top',
      'left',
      'width',
      'height'
    ]
    ),
    (l()(),import0.ɵted((null as any),['\n                                         '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),2,(null as any),View_DayViewComponent_20)),
    import0.ɵdid(270336,(null as any),0,import1.NgTemplateOutlet,[import0.ViewContainerRef],{
      ngTemplateOutlet: [
        0,
        'ngTemplateOutlet'
      ]
      ,
      ngOutletContext: [
        1,
        'ngOutletContext'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['displayEvent']),
    (l()(),import0.ɵted((null as any),['\n                                    ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = ck(v,2,0,(((37 * v.context.$implicit.startOffset) / co.hourParts) + 'px'),(((100 / v.context.$implicit.overlapNumber) * v.context.$implicit.position) + '%'),((100 / v.context.$implicit.overlapNumber) + '%'),((37 * ((v.context.$implicit.endIndex - v.context.$implicit.startIndex) - ((v.context.$implicit.endOffset + v.context.$implicit.startOffset) / co.hourParts))) + 'px'));
    ck(v,1,0,currVal_0);
    const currVal_1:any = co.dayviewNormalEventTemplate;
    const currVal_2:any = ck(v,6,0,v.context.$implicit);
    ck(v,5,0,currVal_1,currVal_2);
  },(null as any));
}
function View_DayViewComponent_18(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),6,'div',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import1.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngClass: [
        0,
        'ngClass'
      ]
    },(null as any)),
    import0.ɵpod(['calendar-event-wrap']),
    (l()(),import0.ɵted((null as any),['\n                                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_DayViewComponent_19)),
    import0.ɵdid(401408,(null as any),0,import1.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                                ']))
  ]
  ,(ck,v) => {
    const currVal_0:any = ck(v,2,0,(<any>v.parent).context.$implicit.events);
    ck(v,1,0,currVal_0);
    const currVal_1:any = (<any>v.parent).context.$implicit.events;
    ck(v,5,0,currVal_1);
  },(null as any));
}
function View_DayViewComponent_17(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),10,'tr',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'td',[[
        'class',
        'calendar-hour-column text-center'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '\n                                ',
      '\n                            '
    ]
    )),
    (l()(),import0.ɵted((null as any),['\n                            '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),4,'td',[
      [
        'class',
        'calendar-cell'
      ]
      ,
      [
        'tappable',
        ''
      ]

    ]
      ,(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.select(v.context.$implicit.time,v.context.$implicit.events)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_DayViewComponent_18)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                            '])),
    (l()(),import0.ɵted((null as any),['\n                        ']))
  ]
  ,(ck,v) => {
    const currVal_1:any = v.context.$implicit.events;
    ck(v,8,0,currVal_1);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.hourColumnLabels[v.context.index];
    ck(v,3,0,currVal_0);
  });
}
function View_DayViewComponent_16(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),11,'init-position-scroll',[[
        'class',
        'dayview-normal-event-container'
      ]
      ],(null as any),[[
        (null as any),
        'onScroll'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('onScroll' === en)) {
        const pd_0:any = ((<any>co.setScrollPosition($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },import2.View_initPositionScrollComponent_0,import2.RenderType_initPositionScrollComponent)),
    import0.ɵdid(2449408,(null as any),0,import3.initPositionScrollComponent,[import0.ElementRef],{
      initPosition: [
        0,
        'initPosition'
      ]
      ,
      emitEvent: [
        1,
        'emitEvent'
      ]

    }
    ,{onScroll: 'onScroll'}),
    (l()(),import0.ɵted(0,['\n                    '])),
      (l()(),import0.ɵeld(0,(null as any),0,7,'table',[[
        'class',
        'table table-bordered table-fixed dayview-normal-event-table'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),4,'tbody',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_DayViewComponent_17)),
    import0.ɵdid(401408,(null as any),0,import1.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵted(0,['\n                ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.initScrollPosition;
    const currVal_1:any = co.preserveScrollPosition;
    ck(v,1,0,currVal_0,currVal_1);
    const currVal_2:any = co.views[1].rows;
    ck(v,8,0,currVal_2);
  },(null as any));
}
function View_DayViewComponent_22(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),7,'tr',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'td',[[
        'class',
        'calendar-hour-column text-center'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '\n                                ',
      '\n                            '
    ]
    )),
    (l()(),import0.ɵted((null as any),['\n                            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'td',[[
        'class',
        'calendar-cell'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                            '])),
    (l()(),import0.ɵted((null as any),['\n                        ']))
  ]
  ,(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.hourColumnLabels[v.context.index];
    ck(v,3,0,currVal_0);
  });
}
function View_DayViewComponent_21(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),11,'init-position-scroll',[[
        'class',
        'dayview-normal-event-container'
      ]
    ],(null as any),(null as any),(null as any),import2.View_initPositionScrollComponent_0,import2.RenderType_initPositionScrollComponent)),
      import0.ɵdid(2449408,(null as any),0,import3.initPositionScrollComponent,[import0.ElementRef],{initPosition: [
        0,
        'initPosition'
      ]
    },(null as any)),
    (l()(),import0.ɵted(0,['\n                    '])),
      (l()(),import0.ɵeld(0,(null as any),0,7,'table',[[
        'class',
        'table table-bordered table-fixed dayview-normal-event-table'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),4,'tbody',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_DayViewComponent_22)),
    import0.ɵdid(401408,(null as any),0,import1.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵted(0,['\n                ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.initScrollPosition;
    ck(v,1,0,currVal_0);
    const currVal_1:any = co.views[1].rows;
    ck(v,8,0,currVal_1);
  },(null as any));
}
function View_DayViewComponent_25(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[(l()(),import0.ɵted((null as any),['\n                                        ']))],(null as any),(null as any));
}
function View_DayViewComponent_24(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),7,'div',[
      [
        'class',
        'calendar-event'
      ]
      ,
      [
        'tappable',
        ''
      ]

    ]
      ,(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.eventSelected(v.context.$implicit.event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import1.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    import0.ɵpod([
      'top',
      'width',
      'height'
    ]
    ),
    (l()(),import0.ɵted((null as any),['\n                                        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),2,(null as any),View_DayViewComponent_25)),
    import0.ɵdid(270336,(null as any),0,import1.NgTemplateOutlet,[import0.ViewContainerRef],{
      ngTemplateOutlet: [
        0,
        'ngTemplateOutlet'
      ]
      ,
      ngOutletContext: [
        1,
        'ngOutletContext'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['displayEvent']),
    (l()(),import0.ɵted((null as any),['\n                                    ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = ck(v,2,0,((25 * v.context.index) + 'px'),'100%','25px');
    ck(v,1,0,currVal_0);
    const currVal_1:any = co.dayviewAllDayEventTemplate;
    const currVal_2:any = ck(v,6,0,v.context.$implicit);
    ck(v,5,0,currVal_1,currVal_2);
  },(null as any));
}
function View_DayViewComponent_23(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),8,'td',[[
        'class',
        'calendar-cell'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import1.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['calendar-event-wrap']),
    import0.ɵdid(139264,(null as any),0,import1.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    import0.ɵpod(['height']),
    (l()(),import0.ɵted((null as any),['\n                                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_DayViewComponent_24)),
    import0.ɵdid(401408,(null as any),0,import1.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                                ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = 'calendar-cell';
    const currVal_1:any = ck(v,2,0,(co.views[2].allDayEvents.length > 0));
    ck(v,1,0,currVal_0,currVal_1);
    const currVal_2:any = ck(v,4,0,((25 * co.views[2].allDayEvents.length) + 'px'));
    ck(v,3,0,currVal_2);
    const currVal_3:any = co.views[2].allDayEvents;
    ck(v,7,0,currVal_3);
  },(null as any));
}
function View_DayViewComponent_26(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'td',[[
        'class',
        'calendar-cell'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                                ']))
  ]
  ,(null as any),(null as any));
}
function View_DayViewComponent_31(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[(l()(),import0.ɵted((null as any),['\n                                         ']))],(null as any),(null as any));
}
function View_DayViewComponent_30(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),7,'div',[
      [
        'class',
        'calendar-event'
      ]
      ,
      [
        'tappable',
        ''
      ]

    ]
      ,(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.eventSelected(v.context.$implicit.event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import1.NgStyle,[
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngStyle: [
        0,
        'ngStyle'
      ]
    },(null as any)),
    import0.ɵpod([
      'top',
      'left',
      'width',
      'height'
    ]
    ),
    (l()(),import0.ɵted((null as any),['\n                                         '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),2,(null as any),View_DayViewComponent_31)),
    import0.ɵdid(270336,(null as any),0,import1.NgTemplateOutlet,[import0.ViewContainerRef],{
      ngTemplateOutlet: [
        0,
        'ngTemplateOutlet'
      ]
      ,
      ngOutletContext: [
        1,
        'ngOutletContext'
      ]

    }
    ,(null as any)),
    import0.ɵpod(['displayEvent']),
    (l()(),import0.ɵted((null as any),['\n                                    ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = ck(v,2,0,(((37 * v.context.$implicit.startOffset) / co.hourParts) + 'px'),(((100 / v.context.$implicit.overlapNumber) * v.context.$implicit.position) + '%'),((100 / v.context.$implicit.overlapNumber) + '%'),((37 * ((v.context.$implicit.endIndex - v.context.$implicit.startIndex) - ((v.context.$implicit.endOffset + v.context.$implicit.startOffset) / co.hourParts))) + 'px'));
    ck(v,1,0,currVal_0);
    const currVal_1:any = co.dayviewNormalEventTemplate;
    const currVal_2:any = ck(v,6,0,v.context.$implicit);
    ck(v,5,0,currVal_1,currVal_2);
  },(null as any));
}
function View_DayViewComponent_29(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),6,'div',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(139264,(null as any),0,import1.NgClass,[
      import0.IterableDiffers,
      import0.KeyValueDiffers,
      import0.ElementRef,
      import0.Renderer
    ]
      ,{ngClass: [
        0,
        'ngClass'
      ]
    },(null as any)),
    import0.ɵpod(['calendar-event-wrap']),
    (l()(),import0.ɵted((null as any),['\n                                    '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_DayViewComponent_30)),
    import0.ɵdid(401408,(null as any),0,import1.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                                ']))
  ]
  ,(ck,v) => {
    const currVal_0:any = ck(v,2,0,(<any>v.parent).context.$implicit.events);
    ck(v,1,0,currVal_0);
    const currVal_1:any = (<any>v.parent).context.$implicit.events;
    ck(v,5,0,currVal_1);
  },(null as any));
}
function View_DayViewComponent_28(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),10,'tr',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'td',[[
        'class',
        'calendar-hour-column text-center'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '\n                                ',
      '\n                            '
    ]
    )),
    (l()(),import0.ɵted((null as any),['\n                            '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),4,'td',[
      [
        'class',
        'calendar-cell'
      ]
      ,
      [
        'tappable',
        ''
      ]

    ]
      ,(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.select(v.context.$implicit.time,v.context.$implicit.events)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_DayViewComponent_29)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                            '])),
    (l()(),import0.ɵted((null as any),['\n                        ']))
  ]
  ,(ck,v) => {
    const currVal_1:any = v.context.$implicit.events;
    ck(v,8,0,currVal_1);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.hourColumnLabels[v.context.index];
    ck(v,3,0,currVal_0);
  });
}
function View_DayViewComponent_27(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),11,'init-position-scroll',[[
        'class',
        'dayview-normal-event-container'
      ]
      ],(null as any),[[
        (null as any),
        'onScroll'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('onScroll' === en)) {
        const pd_0:any = ((<any>co.setScrollPosition($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },import2.View_initPositionScrollComponent_0,import2.RenderType_initPositionScrollComponent)),
    import0.ɵdid(2449408,(null as any),0,import3.initPositionScrollComponent,[import0.ElementRef],{
      initPosition: [
        0,
        'initPosition'
      ]
      ,
      emitEvent: [
        1,
        'emitEvent'
      ]

    }
    ,{onScroll: 'onScroll'}),
    (l()(),import0.ɵted(0,['\n                    '])),
      (l()(),import0.ɵeld(0,(null as any),0,7,'table',[[
        'class',
        'table table-bordered table-fixed dayview-normal-event-table'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),4,'tbody',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_DayViewComponent_28)),
    import0.ɵdid(401408,(null as any),0,import1.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵted(0,['\n                ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.initScrollPosition;
    const currVal_1:any = co.preserveScrollPosition;
    ck(v,1,0,currVal_0,currVal_1);
    const currVal_2:any = co.views[2].rows;
    ck(v,8,0,currVal_2);
  },(null as any));
}
function View_DayViewComponent_33(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),7,'tr',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'td',[[
        'class',
        'calendar-hour-column text-center'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '\n                                ',
      '\n                            '
    ]
    )),
    (l()(),import0.ɵted((null as any),['\n                            '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'td',[[
        'class',
        'calendar-cell'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                            '])),
    (l()(),import0.ɵted((null as any),['\n                        ']))
  ]
  ,(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.hourColumnLabels[v.context.index];
    ck(v,3,0,currVal_0);
  });
}
function View_DayViewComponent_32(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),11,'init-position-scroll',[[
        'class',
        'dayview-normal-event-container'
      ]
    ],(null as any),(null as any),(null as any),import2.View_initPositionScrollComponent_0,import2.RenderType_initPositionScrollComponent)),
      import0.ɵdid(2449408,(null as any),0,import3.initPositionScrollComponent,[import0.ElementRef],{initPosition: [
        0,
        'initPosition'
      ]
    },(null as any)),
    (l()(),import0.ɵted(0,['\n                    '])),
      (l()(),import0.ɵeld(0,(null as any),0,7,'table',[[
        'class',
        'table table-bordered table-fixed dayview-normal-event-table'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),4,'tbody',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_DayViewComponent_33)),
    import0.ɵdid(401408,(null as any),0,import1.NgForOf,[
      import0.ViewContainerRef,
      import0.TemplateRef,
      import0.IterableDiffers
    ]
      ,{ngForOf: [
        0,
        'ngForOf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵted(0,['\n                ']))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.initScrollPosition;
    ck(v,1,0,currVal_0);
    const currVal_1:any = co.views[2].rows;
    ck(v,8,0,currVal_1);
  },(null as any));
}
export function View_DayViewComponent_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    import0.ɵqud(201326592,1,{slider: 0}),
    (l()(),import0.ɵted((null as any),['\n        '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),107,'ion-slides',([] as any[]),(null as any),[[
        (null as any),
        'ionSlideDidChange'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import4.DayViewComponent = v.component;
      if (('ionSlideDidChange' === en)) {
        const pd_0:any = ((<any>co.onSlideChanged()) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },import5.View_Slides_0,import5.RenderType_Slides)),
    import0.ɵdid(614400,[
      [
        1,
        4
      ]
      ,
      [
        'daySlider',
        4
      ]

    ]
    ,0,import6.Slides,[
      import7.Config,
      import8.Platform,
      import0.NgZone,
      [
        2,
        import9.ViewController
      ]
      ,
      import0.ElementRef,
      import0.Renderer
    ]
    ,{
      loop: [
        0,
        'loop'
      ]
      ,
      dir: [
        1,
        'dir'
      ]
      ,
      spaceBetween: [
        2,
        'spaceBetween'
      ]

    }
    ,{ionSlideDidChange: 'ionSlideDidChange'}),
    (l()(),import0.ɵted(0,['\n            '])),
    (l()(),import0.ɵeld(0,(null as any),0,33,'ion-slide',([] as any[]),(null as any),(null as any),(null as any),import10.View_Slide_0,import10.RenderType_Slide)),
    import0.ɵdid(90112,(null as any),0,import11.Slide,[
      import0.ElementRef,
      import0.Renderer,
      import6.Slides
    ]
    ,(null as any),(null as any)),
    (l()(),import0.ɵted(0,['\n                '])),
      (l()(),import0.ɵeld(0,(null as any),0,23,'div',[[
        'class',
        'dayview-allday-table'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',[[
        'class',
        'dayview-allday-label'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    )),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),17,'ion-scroll',[
      [
        'class',
        'dayview-allday-content-wrapper'
      ]
      ,
      [
        'scrollY',
        'true'
      ]
      ,
      [
        'zoom',
        'false'
      ]

    ]
    ,[
      [
        2,
        'scroll-x',
        (null as any)
      ]
      ,
      [
        2,
        'scroll-y',
        (null as any)
      ]

    ]
    ,(null as any),(null as any),import12.View_Scroll_0,import12.RenderType_Scroll)),
    import0.ɵdid(24576,(null as any),0,import13.Scroll,[import0.ElementRef],{
      scrollY: [
        0,
        'scrollY'
      ]
      ,
      zoom: [
        1,
        'zoom'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted(0,['\n                        '])),
      (l()(),import0.ɵeld(0,(null as any),0,13,'table',[[
        'class',
        'table table-bordered dayview-allday-content-table'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                            '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),10,'tbody',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                            '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),7,'tr',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_DayViewComponent_1)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_DayViewComponent_4)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                            '])),
    (l()(),import0.ɵted((null as any),['\n                            '])),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵted(0,['\n                    '])),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵted(0,['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),0,1,(null as any),View_DayViewComponent_5)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted(0,['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),0,1,(null as any),View_DayViewComponent_10)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted(0,['\n            '])),
    (l()(),import0.ɵted(0,['\n            '])),
    (l()(),import0.ɵeld(0,(null as any),0,33,'ion-slide',([] as any[]),(null as any),(null as any),(null as any),import10.View_Slide_0,import10.RenderType_Slide)),
    import0.ɵdid(90112,(null as any),0,import11.Slide,[
      import0.ElementRef,
      import0.Renderer,
      import6.Slides
    ]
    ,(null as any),(null as any)),
    (l()(),import0.ɵted(0,['\n                '])),
      (l()(),import0.ɵeld(0,(null as any),0,23,'div',[[
        'class',
        'dayview-allday-table'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',[[
        'class',
        'dayview-allday-label'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    )),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),17,'ion-scroll',[
      [
        'class',
        'dayview-allday-content-wrapper'
      ]
      ,
      [
        'scrollY',
        'true'
      ]
      ,
      [
        'zoom',
        'false'
      ]

    ]
    ,[
      [
        2,
        'scroll-x',
        (null as any)
      ]
      ,
      [
        2,
        'scroll-y',
        (null as any)
      ]

    ]
    ,(null as any),(null as any),import12.View_Scroll_0,import12.RenderType_Scroll)),
    import0.ɵdid(24576,(null as any),0,import13.Scroll,[import0.ElementRef],{
      scrollY: [
        0,
        'scrollY'
      ]
      ,
      zoom: [
        1,
        'zoom'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted(0,['\n                        '])),
      (l()(),import0.ɵeld(0,(null as any),0,13,'table',[[
        'class',
        'table table-bordered dayview-allday-content-table'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                            '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),10,'tbody',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                            '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),7,'tr',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_DayViewComponent_12)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_DayViewComponent_15)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                            '])),
    (l()(),import0.ɵted((null as any),['\n                            '])),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵted(0,['\n                    '])),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵted(0,['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),0,1,(null as any),View_DayViewComponent_16)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted(0,['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),0,1,(null as any),View_DayViewComponent_21)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted(0,['\n            '])),
    (l()(),import0.ɵted(0,['\n            '])),
    (l()(),import0.ɵeld(0,(null as any),0,33,'ion-slide',([] as any[]),(null as any),(null as any),(null as any),import10.View_Slide_0,import10.RenderType_Slide)),
    import0.ɵdid(90112,(null as any),0,import11.Slide,[
      import0.ElementRef,
      import0.Renderer,
      import6.Slides
    ]
    ,(null as any),(null as any)),
    (l()(),import0.ɵted(0,['\n                '])),
      (l()(),import0.ɵeld(0,(null as any),0,23,'div',[[
        'class',
        'dayview-allday-table'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                    '])),
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',[[
        'class',
        'dayview-allday-label'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),[
      '',
      ''
    ]
    )),
    (l()(),import0.ɵted((null as any),['\n                    '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),17,'ion-scroll',[
      [
        'class',
        'dayview-allday-content-wrapper'
      ]
      ,
      [
        'scrollY',
        'true'
      ]
      ,
      [
        'zoom',
        'false'
      ]

    ]
    ,[
      [
        2,
        'scroll-x',
        (null as any)
      ]
      ,
      [
        2,
        'scroll-y',
        (null as any)
      ]

    ]
    ,(null as any),(null as any),import12.View_Scroll_0,import12.RenderType_Scroll)),
    import0.ɵdid(24576,(null as any),0,import13.Scroll,[import0.ElementRef],{
      scrollY: [
        0,
        'scrollY'
      ]
      ,
      zoom: [
        1,
        'zoom'
      ]

    }
    ,(null as any)),
    (l()(),import0.ɵted(0,['\n                        '])),
      (l()(),import0.ɵeld(0,(null as any),0,13,'table',[[
        'class',
        'table table-bordered dayview-allday-content-table'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                            '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),10,'tbody',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                            '])),
    (l()(),import0.ɵeld(0,(null as any),(null as any),7,'tr',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵted((null as any),['\n                                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_DayViewComponent_23)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                                '])),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_DayViewComponent_26)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted((null as any),['\n                            '])),
    (l()(),import0.ɵted((null as any),['\n                            '])),
    (l()(),import0.ɵted((null as any),['\n                        '])),
    (l()(),import0.ɵted(0,['\n                    '])),
    (l()(),import0.ɵted((null as any),['\n                '])),
    (l()(),import0.ɵted(0,['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),0,1,(null as any),View_DayViewComponent_27)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted(0,['\n                '])),
    (l()(),import0.ɵand(8388608,(null as any),0,1,(null as any),View_DayViewComponent_32)),
    import0.ɵdid(8192,(null as any),0,import1.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    (l()(),import0.ɵted(0,['\n            '])),
    (l()(),import0.ɵted(0,['\n        '])),
    (l()(),import0.ɵted((null as any),['\n    ']))
  ]
  ,(ck,v) => {
    var co:import4.DayViewComponent = v.component;
    const currVal_0:any = true;
    const currVal_1:any = co.dir;
    const currVal_2:any = co.spaceBetween;
    ck(v,3,0,currVal_0,currVal_1,currVal_2);
    const currVal_6:any = 'true';
    const currVal_7:any = 'false';
    ck(v,14,0,currVal_6,currVal_7);
    const currVal_8:any = (0 === co.currentViewIndex);
    ck(v,23,0,currVal_8);
    const currVal_9:any = (0 !== co.currentViewIndex);
    ck(v,26,0,currVal_9);
    const currVal_10:any = (0 === co.currentViewIndex);
    ck(v,34,0,currVal_10);
    const currVal_11:any = (0 !== co.currentViewIndex);
    ck(v,37,0,currVal_11);
    const currVal_15:any = 'true';
    const currVal_16:any = 'false';
    ck(v,49,0,currVal_15,currVal_16);
    const currVal_17:any = (1 === co.currentViewIndex);
    ck(v,58,0,currVal_17);
    const currVal_18:any = (1 !== co.currentViewIndex);
    ck(v,61,0,currVal_18);
    const currVal_19:any = (1 === co.currentViewIndex);
    ck(v,69,0,currVal_19);
    const currVal_20:any = (1 !== co.currentViewIndex);
    ck(v,72,0,currVal_20);
    const currVal_24:any = 'true';
    const currVal_25:any = 'false';
    ck(v,84,0,currVal_24,currVal_25);
    const currVal_26:any = (2 === co.currentViewIndex);
    ck(v,93,0,currVal_26);
    const currVal_27:any = (2 !== co.currentViewIndex);
    ck(v,96,0,currVal_27);
    const currVal_28:any = (2 === co.currentViewIndex);
    ck(v,104,0,currVal_28);
    const currVal_29:any = (2 !== co.currentViewIndex);
    ck(v,107,0,currVal_29);
  },(ck,v) => {
    var co:import4.DayViewComponent = v.component;
    const currVal_3:any = co.allDayLabel;
    ck(v,11,0,currVal_3);
    const currVal_4:any = import0.ɵnov(v,14).scrollX;
    const currVal_5:any = import0.ɵnov(v,14).scrollY;
    ck(v,13,0,currVal_4,currVal_5);
    const currVal_12:any = co.allDayLabel;
    ck(v,46,0,currVal_12);
    const currVal_13:any = import0.ɵnov(v,49).scrollX;
    const currVal_14:any = import0.ɵnov(v,49).scrollY;
    ck(v,48,0,currVal_13,currVal_14);
    const currVal_21:any = co.allDayLabel;
    ck(v,81,0,currVal_21);
    const currVal_22:any = import0.ɵnov(v,84).scrollX;
    const currVal_23:any = import0.ɵnov(v,84).scrollY;
    ck(v,83,0,currVal_22,currVal_23);
  });
}
function View_DayViewComponent_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'dayview',([] as any[]),[[
        2,
        'dayview',
        (null as any)
      ]
    ],(null as any),(null as any),View_DayViewComponent_0,RenderType_DayViewComponent)),
    import0.ɵdid(2482176,(null as any),0,import4.DayViewComponent,[
      import14.CalendarService,
      import0.ElementRef
    ]
    ,(null as any),(null as any))
  ]
  ,(ck,v) => {
    ck(v,1,0);
  },(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,1).class;
    ck(v,0,0,currVal_0);
  });
}
export const DayViewComponentNgFactory:import0.ComponentFactory<import4.DayViewComponent> = import0.ɵccf('dayview',import4.DayViewComponent,View_DayViewComponent_Host_0,{
  dayviewAllDayEventTemplate: 'dayviewAllDayEventTemplate',
  dayviewNormalEventTemplate: 'dayviewNormalEventTemplate',
  formatHourColumn: 'formatHourColumn',
  formatDayTitle: 'formatDayTitle',
  allDayLabel: 'allDayLabel',
  hourParts: 'hourParts',
  eventSource: 'eventSource',
  markDisabled: 'markDisabled',
  locale: 'locale',
  dateFormatter: 'dateFormatter',
  dir: 'dir',
  scrollToHour: 'scrollToHour',
  preserveScrollPosition: 'preserveScrollPosition',
  lockSwipeToPrev: 'lockSwipeToPrev',
  lockSwipes: 'lockSwipes',
  startHour: 'startHour',
  endHour: 'endHour',
  spaceBetween: 'spaceBetween'
}
,{
  onRangeChanged: 'onRangeChanged',
  onEventSelected: 'onEventSelected',
  onTimeSelected: 'onTimeSelected',
  onTitleChanged: 'onTitleChanged'
}
,([] as any[]));
